bowling.js

```diff
@@ -0,0 +1,51 @@
+function BowlingGame() {
+  this.rolls = [];
+}
+
+BowlingGame.prototype.isSpare = function(roll1, roll2) {
+  return roll1 + roll2 === 10;
+}
+BowlingGame.prototype.isStrike = function(roll){
+  return roll === 10;
+}
+
+BowlingGame.prototype.calculateSpareBonus = function(frameIndex){
+  return this.rolls[frameIndex + 2];
+}
+BowlingGame.prototype.calculateStrikeBonus = function(frameIndex){
+  if(frameIndex !== 19){
+    if(this.rolls[frameIndex + 2] === 10){
+      //if the next roll is also a strike, for bonus add rolls from the next two frames
+      return this.rolls[frameIndex + 2] + this.rolls[frameIndex + 4];
+    } else {
+      //if the next roll is not strike add the two rolls in that frame
+      return this.rolls[frameIndex + 2] + this.rolls[frameIndex + 3];
+    }
+  }
+}
+BowlingGame.prototype.roll = function(pins) {
```

Comment:
the Roll method is doing too much - it should just roll, it should not be checking the number of pins.

Tip: This can be a single line function